pkg/controllers/kubeconfig/watch_predicates.go

```diff
@@ -0,0 +1,56 @@
+package kubeconfig
+
+import (
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/openshift/cluster-capi-operator/pkg/controllers"
+)
+
+func toTokenSecret(client.Object) []reconcile.Request {
+	return []reconcile.Request{{
+		NamespacedName: client.ObjectKey{Name: tokenSecretName, Namespace: controllers.DefaultManagedNamespace},
+	}}
+}
+
+func tokenSecretPredicate() predicate.Funcs {
+	isOwnedTokenSecret := func(obj runtime.Object) bool {
+		secret, ok := obj.(*corev1.Secret)
+		if !ok {
+			panic("expected to get an of object of type corev1.Secret")
+		}
+
+		return secret.GetNamespace() == controllers.DefaultManagedNamespace && secret.GetName() == tokenSecretName
+	}
+
+	return predicate.Funcs{
+		CreateFunc:  func(e event.CreateEvent) bool { return isOwnedTokenSecret(e.Object) },
+		UpdateFunc:  func(e event.UpdateEvent) bool { return isOwnedTokenSecret(e.ObjectNew) },
+		DeleteFunc:  func(e event.DeleteEvent) bool { return isOwnedTokenSecret(e.Object) },
+		GenericFunc: func(e event.GenericEvent) bool { return isOwnedTokenSecret(e.Object) },
+	}
+}
+
+func kubeconfigSecretPredicate() predicate.Funcs {
+	isKubeconfigSecret := func(obj runtime.Object) bool {
+		secret, ok := obj.(*corev1.Secret)
+		if !ok {
```

Comment:
are you sure that we want the code to panic and not use the pattern we have here https://github.com/openshift/cluster-capi-operator/blob/main/pkg/controllers/secretsync/watch_predicates.go#L21?

Reply:
We decided to panic in these cases. For instance: https://github.com/openshift/cluster-control-plane-machine-set-operator/blob/main/pkg/controllers/controlplanemachineset/watch_filters.go#L56-L59

Here's @JoelSpeed comment: https://github.com/openshift/cluster-control-plane-machine-set-operator/pull/28#discussion_r844859820